src/test/java/com/klodnicki/Bike/v2/service/ChargingStationServiceHandlerTest.java

```diff
@@ -0,0 +1,158 @@
+    @Autowired
+    private ModelMapper modelMapper;
+
+    @Test
+    public void add_ShouldAddToDatabase_WhenGivenCorrectArguments() {
+        //given
+        chargingStationRepository.deleteAll();
+        Long expected = chargingStationRepository.count() + 1;
+        ChargingStation chargingStation = new ChargingStation();
+
+        //when
+        chargingStationServiceHandler.add(chargingStation);
+        Long actual = chargingStationRepository.count();
+
+        //then
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void add_ShouldReturnStationForAdminResponseDTO_WhenGivenChargingStation() {
+        //given
+        chargingStationRepository.deleteAll();
+        ChargingStation chargingStation = new ChargingStation();
+        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+
+        //when
+        StationForAdminResponseDTO actual = chargingStationServiceHandler.add(chargingStation);
+
+        //then
+        assertNotNull(actual.getId());
+        expected.setId(actual.getId());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void findAll_ShouldReturnListOfStationForAdminResponseDTO_WhenExistInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+        List<StationForAdminResponseDTO> expectedStationDTOS = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            ChargingStation chargingStation = new ChargingStation();
+            chargingStationRepository.save(chargingStation);
+            StationForAdminResponseDTO stationDTO = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+            expectedStationDTOS.add(stationDTO);
+        }
+
+        //when
+        List<StationForAdminResponseDTO> actual = chargingStationServiceHandler.findAll();
+
+        //then
+        assertEquals(expectedStationDTOS, actual);
+    }
+
+    @Test
+    public void findById_ShouldReturnStationForAdminResponseDTO_WhenExistInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+        ChargingStation chargingStation = new ChargingStation();
+        chargingStationRepository.save(chargingStation);
+        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+
+        //when
+        StationForAdminResponseDTO actual = chargingStationServiceHandler.findById(chargingStation.getId());
+
+        //then
+        assertEquals(expected, actual);
+    }
+
+
+    @Test
+    public void findStationById_ShouldReturnChargingStation_WhenExistInDatabase() {
+        //given
+        chargingStationRepository.deleteAll();
+        ChargingStation chargingStation = new ChargingStation();
+        chargingStationRepository.save(chargingStation);
+
+        //when
+        ChargingStation actual = chargingStationServiceHandler.findStationById(chargingStation.getId());
+        actual.setBikeList(null);
```

Comment:
Is this line necessary here?

Reply:
yes, otherwise first object will have a null in bikeList and second object will have an empty list [] which is not the same and test will fail

Reply:
Yup, those are not the same. You should look why is this happening and change accordingly – for example in the `//given` phase of the test. 

If you manually change outcome of the **actual** object you're cheating a little bit. Like on an exam :) That's because you should test it without altering the object further. If this method later break and i.e. return a list with one completely random bike the test won't alert it, because the actual object is altered later.

Reply:
understood! corrected and ready for re-review!